### ISSUE TITLE
NSError is imported as non-optional in swift in didCompletePasswordAuthenticationStepWithError

### ISSUE THREAD

**Issue by jawwad** [aws/aws-sdk-ios#433]
In AWSCognitoIdentityUserPool.h, the signature of the `didCompletePasswordAuthenticationStepWithError` method is the following:

```objc
-(void) didCompletePasswordAuthenticationStepWithError:(NSError *) error;
```

Since the `NS_ASSUME_NONNULL_BEGIN` and `NS_ASSUME_NONNULL_END` specifiers are used the `_Nullable` attribute needs to be added to `NSError` otherwise in swift the error parameter cannot be checked to see if it is nil since it is imported as a non-optional.

![image](https://cloud.githubusercontent.com/assets/202240/16987016/53101418-4ea3-11e6-82ed-7d79438226aa.png)

The method signature can be forced to use NSError? but that results in a warning:
![image](https://cloud.githubusercontent.com/assets/202240/16987053/89725016-4ea3-11e6-9573-4ab0ed1d1878.png)

Updating the signature of the method in `AWSCognitoIdentityUserPool.h` to the following should fix it:

```objc
- (void)didCompletePasswordAuthenticationStepWithError:(NSError * _Nullable) error;
```

**Comment by karthiksaligrama** [comment]
Thanks , we will fix this in the next release.

**Comment by karthiksaligrama** [comment]
Fixed with 2.4.6
